azurelinuxagent/agent.py

```diff
@@ -151,6 +151,23 @@ def show_configuration(self):
         for k in sorted(configuration.keys()):
             print("{0} = {1}".format(k, configuration[k]))
 
+    def collect_logs(self, log_collector_mode):
+        from azurelinuxagent.common.logcollector import LogCollector, OUTPUT_RESULTS_FILE_PATH
+        if log_collector_mode == "full":
```

Comment:
Would it be useful to make this string a constant? I'm mainly concerned with additional modes being added and the potential therein to introduce typos, especially because this string is repeated down below (line 162).

Reply:
+1 to this, it would be helpful to have an Enum or something here just to make this a bit more clearer as to what this is doing and what the multiple options are.

Reply:
The log collector is modelled after IID, which has two modes as well which have been sufficient for our needs, so I don't expect other modes to be added here, to be honest. On the other hand, the only valid option for `log_collector_mode` is "full". If the parameter isn't specified, normal mode will be used. I can make the "full" command-line option as a constant to make it cleaner, though.